app/src/main/java/bitcoin/wallet/modules/transactions/TransactionsInteractor.kt

```diff
@@ -60,10 +60,40 @@ class TransactionsInteractor(private val adapterManager: AdapterManager, private
                         confirmations = TransactionRecordViewItem.getConfirmationsCount(record.blockHeight, adapter.latestBlockHeight)
                 )
                 items.add(item)
+                record.timestamp?.let { timestamp ->
+                    flowableList.add(
+                            exchangeRateManager.getRate(coinCode = record.coinCode, currency = DollarCurrency().code, timestamp = timestamp)
+                                    .map { Pair(record.transactionHash, it) }
+                    )
+                }
             }
         }
 
-        delegate?.didRetrieveItems(items)
+        items.sortByDescending { it.date }
+
+        disposables.add(Flowable.zip(flowableList, Arrays::asList)
+                .subscribeOn(Schedulers.io())
+                .unsubscribeOn(Schedulers.io())
+                .observeOn(io.reactivex.android.schedulers.AndroidSchedulers.mainThread())
+                .subscribe { resultRates ->
+                    val ratesMap = mutableMapOf<String, Double>()
```

Comment:
Please check if there is a shorter way for converting list of pairs to map

Reply:
I couldn't find shorter version, maybe you can help me